state/account.go

```diff
@@ -93,13 +129,77 @@ func (st *Account) Deserialize(buf []byte) error {
 	return nil
 }
 
+// IsNewbieAccount returns true if the account has not sent any actions
+func (st *Account) IsNewbieAccount() bool {
+	return st.nonce == 0
+}
+
+// AccountType returns the account type
+func (st *Account) AccountType() int32 {
+	return st.accountType
+}
+
+// SetPendingNonce sets the pending nonce
+func (st *Account) SetPendingNonce(nonce uint64) error {
+	switch st.accountType {
+	case 1:
+		if nonce != st.nonce+1 {
+			return errors.Wrapf(ErrInvalidNonce, "actual value %d, %d expected", nonce, st.nonce+1)
+		}
+		st.nonce++
+	case 0:
+		if nonce != st.nonce+2 {
+			return errors.Wrapf(ErrInvalidNonce, "actual value %d, %d expected", nonce, st.nonce+2)
+		}
+		st.nonce++
+	default:
+		return errors.Wrapf(ErrUnknownAccountType, "account type %d", st.accountType)
+	}
+
+	return nil
+}
+
+// PendingNonce returns the pending nonce of the account
+func (st *Account) PendingNonce() uint64 {
+	switch st.accountType {
+	case 1:
+		return st.nonce
```

Comment:
The nonce concept in account struct is ambiguous, where one is confirmed nonce whereas another is pendingnonce. So it had better be separated into two entites, both of whom implements account interface.

Reply:
You may try to do it in a follow up PR, and we can wait for your PR before the new cut. 
In current implementation, `nonce` is a private variable. `PendingNonce` and `SetPendingNonce` are the functions for external usage. I think it is clear.